### PR TITLE
uftrace: Add 'void' to functions without arguments

### PR DESCRIPTION
**Pull request by paranlee**
This patch remove argument on void non-args function when using
`fstack_get_loc_mode()`, fstack_get_filter_mode()`.

Fixed: #1719
Signed-off-by: Paran Lee <p4ranlee@gmail.com>

### CODE REVIEW

**Review by namhyung — COMMENTED** [review]
OK, we have a lot of such functions.  I think you can make the subject in the second commit more compact like "uftrace: Add 'void' to functions without arguments".  You can use 'uftrace:' prefix when it doesn't belong to a specific area.

Also can we check this void argument before merging the code?  I think we can add a warning in the Makefile or update the clang-format check.  Thoughts?

**Comment by paranlee** [comment]
@namhyung  I add compile options at Makefile. I tested with [without void](https://github.com/namhyung/uftrace/issues/1719#issuecomment-1575558281) at `DEBUG=1`

File: Makefile [hunk 1]
```diff
@@ -104,7 +104,7 @@ LIB_LDFLAGS        = $(COMMON_LDFLAGS) $(LDFLAGS_$@) $(LDFLAGS_lib) -Wl,--no-und
 TEST_LDFLAGS       = $(COMMON_LDFLAGS) -L$(objdir)/libtraceevent -ltraceevent
 
 ifeq ($(DEBUG), 1)
-  COMMON_CFLAGS += -O0 -g3 -DDEBUG_MODE=1
+  COMMON_CFLAGS += -O0 -g3 -DDEBUG_MODE=1 -Werror=strict-prototypes
```
Comment (namhyung) [inline Makefile:107]: Please do it in a separate commit.  Also I think we can add `-Werror` here and add `-Wstrict-prototypes` to all conditions.

Comment (paranlee) [reply Makefile]: OK, Thanks for code review!

**Review by namhyung — COMMENTED** [review]
You don't need to add 'Fixed:' tag to every commit.  Anyway, looks ok.
